src/ucan/internal.ts

```diff
@@ -0,0 +1,123 @@
+import localforage from 'localforage'
+
+import * as common from '../common'
+import * as pathUtil from '../fs/path'
+import * as ucan from '../ucan'
+import { UCANS_STORAGE_KEY } from '../common'
+import { Prerequisites } from './prerequisites'
+import { Ucan, WNFS_PREFIX } from '../ucan'
+import { setup } from '../setup/internal'
+
+
+let dictionary: Record<string, Ucan> = {}
+
+
+// FUNCTIONS
+
+
+/**
+ * You didn't see anything 👀
+ */
+export async function clearStorage(): Promise<void> {
+  dictionary = {}
+  await localforage.removeItem(UCANS_STORAGE_KEY)
+}
+
+/**
+ * Lookup the prefix for a filesystem key in the dictionary.
+ */
+export function dictionaryFilesystemPrefix(username: string): string {
+  // const host = `${username}.${setup.endpoints.user}`
+  // TODO: Waiting on API change.
+  //       Should be `${WNFS_PREFIX}:${host}/`
+  return WNFS_PREFIX + ":/"
+}
+
+/**
+ * Look up a UCAN with a file system path.
+ */
+export async function lookupFilesystemUcan(path: string): Promise<Ucan | null> {
+  const pathParts = pathUtil.splitParts(path)
+  const username = await common.authenticatedUsername()
+  const prefix = username ? dictionaryFilesystemPrefix(username) : ""
+
+  if (dictionary["*"]) {
```

Comment:
Don't think so, since you can have the `*` resource. Prefix would only be necessary if the resource of a UCAN would be an object.

Reply:
Okay yeah I think this makes sense since we're only ever interacting with one FS at a time right now :ok_hand: 

(Hopefully soon!) apps will be able to interact with multiple filesystems. If we store all available UCANs in the same dictionary, we'll need to preface them with the username. If we have a different dictionary per FS this will be fine